sympy/geometry/curve.py

```diff
@@ -246,20 +263,149 @@ def parameter(self):
         return self.args[1][0]
 
     @property
+    def dimension(self):
+        """ The dimension of the curve.
+
+        Returns
+        =======
+
+        dimension : Integer
+
+        Examples
+        ========
+
+        >>> from sympy.abc import t
+        >>> from sympy.geometry import Curve
+        >>> C = Curve([t, t, t], (t, 0, 2))
+        >>> C.dimension
+        3
+        """
+        return len(self.functions)
+
+    @property
     def length(self):
         """The curve length.
 
         Examples
         ========
 
         >>> from sympy.geometry.curve import Curve
-        >>> from sympy import cos, sin
         >>> from sympy.abc import t
         >>> Curve((t, t), (t, 0, 1)).length
         sqrt(2)
         """
-        integrand = sqrt(sum(diff(func, self.limits[0])**2 for func in self.functions))
-        return integrate(integrand, self.limits)
+        return self.functions.diff(self.parameter).norm().integrate(self.limits).simplify()
```

Comment:
Rather than changing the return type of `functions` to Matrix from Tuple, my initial preference would be to only recast the Tuple as Matrix where needed -- like here.

```
return Matrix(self).diff(...
```